### ISSUE TITLE
disable caching in vault-agent configuration 

### ISSUE THREAD

**Issue by Ardiea** [mitodl/ol-infrastructure#1394]
# Description/Context
Starting with vault 1.13.0, vault now support an "api_proxy" stanza in place of "cache". If we were to drop the "cache" stanza from our configuration and replace it with "api_proxy" we can disable the local caching of vault requests across all of our nodes. IMO, this is a good thing. The vault agent is non-blocking, meaning a cached result in the agent won't be updated if something changes @ the server with the actual value, for instance a revocation, or w/e. Caching leads to different perceptions of reality. 

More information: https://developer.hashicorp.com/vault/docs/agent/apiproxy#configuration-api_proxy

**Comment by blarghmatey** [comment]
This depends on #1009. The current mount for Bootcamps database credentials is using a deprecated plugin in Vault. That prevents us from upgrading our clusters past 1.12

**Comment by Ardiea** [comment]
As part of better undstanding this issue, the following:

https://developer.hashicorp.com/vault/docs/v1.15.x/agent-and-proxy#introduce-vault-agent-and-vault-proxy-to-the-workflow

TL;DR: Right now a vault agent acts as a proxy when started with `vault agent` but "some day soon" the proxy functionality will be removed from the agent and our two usages for vault at the moment (templating and proxying) will become divided. We will need to run vault twice on servers that need both sets of functionality 👎 

I think the go-forward would be:
- Stop using vault-template and switch entirely to consul-template for our template needs.
- Convert all our current vault-agent installations into vault-proxy installations.
- Consul-template points to `vault-proxy` exactly the same way it currently points to `vault-agent`, should be a transparent change from that perspective.

**Comment by Ardiea** [comment]
Close as duplicate of #1640
